packages/dashboard-frontend/src/containers/Loader/Factory/Steps/Apply/Devfile/__tests__/index.spec.tsx

```diff
@@ -151,13 +155,224 @@ describe('Factory Loader container, step CREATE_WORKSPACE__APPLYING_DEVFILE', ()
     expect(mockOnNextStep).not.toHaveBeenCalled();
   });
 
+  describe('configure remotes', () => {
+    test('remotes configured with urls', async () => {
+      const store = getStoreBuilder()
+        .withFactoryResolver({
+          resolver: {},
+          converted: {
+            devfileV2: devfile,
+          },
+        })
+        .build();
+
+      const remotesAttr = '{http://git-test-1.git,http://git-test-2.git,http://git-test-3.git}';
+      searchParams.append(REMOTES_ATTR, remotesAttr);
+
+      renderComponent(store, loaderSteps, searchParams);
+      jest.advanceTimersByTime(MIN_STEP_DURATION_MS);
+
+      factoryId = `${REMOTES_ATTR}=${remotesAttr}&` + factoryId;
+
+      await waitFor(() =>
+        expect(prepareDevfile).toHaveBeenCalledWith(devfile, factoryId, undefined, false),
+      );
+
+      expect(devfile.projects).not.toBe(undefined);
+      expect(devfile.projects!.length).toBe(1);
+      expect(devfile.projects![0]).toMatchObject({
```

Comment:
Using the optional chaining operator is safer than the non-null assertion operator. Besides, you will not need to disable the corresponding `eslint` rule to make it happy.

```suggestion
      expect(devfile.projects?.length).toBe(1);
      expect(devfile.projects?.[0]).toMatchObject({
```